src/ynEIGEN/ynEigenViewer.sol

```diff
@@ -1,37 +1,60 @@
 // SPDX-License-Identifier: BSD 3-Clause License
 pragma solidity ^0.8.24;
 
+import {Initializable} from "lib/openzeppelin-contracts-upgradeable/contracts/proxy/utils/Initializable.sol";
 import {IERC20, IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
 import {IwstETH} from "../external/lido/IwstETH.sol";
 import {IynEigen} from "../interfaces/IynEigen.sol";
 import {IRateProvider} from "../interfaces/IRateProvider.sol";
-import {ITokenStakingNodesManager,ITokenStakingNode} from "../interfaces/ITokenStakingNodesManager.sol";
+import {ITokenStakingNodesManager, ITokenStakingNode} from "../interfaces/ITokenStakingNodesManager.sol";
 import {IERC4626} from "@openzeppelin/contracts/interfaces/IERC4626.sol";
 import {AssetRegistry} from "./AssetRegistry.sol";
 import {IEigenStrategyManager} from "../interfaces/IEigenStrategyManager.sol";
 
+contract ynEigenViewer is Initializable {
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  ERRORS  ------------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    error ZeroAddress();
 
-contract ynEigenViewer {
-    
     struct AssetInfo {
         address asset;
         string name;
         string symbol;
         uint256 ratioOfTotalAssets;
         uint256 totalBalance;
     }
-    
-    /* solhint-disable immutable-vars-naming */
-    AssetRegistry public immutable assetRegistry;
-    IynEigen public immutable ynEIGEN;
-    ITokenStakingNodesManager public immutable tokenStakingNodesManager;
-    IRateProvider public immutable rateProvider;
-    /* solhint-enable immutable-vars-naming */
+
+    AssetRegistry public assetRegistry;
+    IynEigen public ynEIGEN;
```

Comment:
let's make sure these changes don't get in, maybe it's a rebase issue

Reply:
fixed in [334c850](https://github.com/yieldnest/yieldnest-protocol/pull/158/commits/334c8506843f919cc8ace346a11a36722d397997)